src/freertos_drivers/ti/SimplelinkPthread.cxx

```diff
@@ -0,0 +1,254 @@
+/** @copyright
+ * Copyright (c) 2018, Stuart W Baker
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ * 
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file SimpleLinkPthread.cxx
+ * This file represents a C language abstraction of common operating
+ * system calls specific to the Simplelink SDK pthread support.
+ *
+ * @author Stuart W. Baker
+ * @date 29 December 2018
+ */
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+
+#include "nmranet_config.h"
+
+#include "os/os.h"
+#include "os/os_private.h"
+
+extern "C"
+{
+#if defined (__FreeRTOS__)
+extern void hw_postinit(void);
+
+/// default stdin
+extern const char *STDIN_DEVICE;
+
+/// default stdout
+extern const char *STDOUT_DEVICE;
+
+/// default stderr
+extern const char *STDERR_DEVICE;
+
+/// Private metadata for starting a thread.
+typedef struct
+{
+    void *(*entry)(void*); ///< thread entry point
+    void *arg; ///< argument to thread
+    TaskList *taskList; //< task list instance
+} OSThreadStartPriv;
+
+/// Entry point to a thread.
+/// @param arg metadata for entering the thread
+static void *os_thread_start(void *arg)
```

Comment:
we decided to platformize this as well with some hooks maybe.

Reply:
Fixed.